src/views/AnalysisModule/Analysis/index.tsx

```diff
@@ -16,16 +32,180 @@ interface ComponentProps {
     title: string;
     startDate?: string;
     endDate?: string;
+    activeProject: number;
+    analysisId: number;
+    setAnalysisDeleteId: () => number;
+}
+
+interface AnalysisPillars {
+    id: number;
+    assigneeName: string;
+    title: string;
+    mainStatement: string;
+    informationGap: string;
+    filters?: string;
+    analysis: number;
+}
+
+interface AnalysisPillarRendererProps extends Omit<AnalysisPillars, 'id'> {
+    pillarId: AnalysisPillars['id'];
+    setDeleteId: () => number;
 }
 
+const mapStateToProps = (state: AppState) => ({
+    activeProject: activeProjectIdFromStateSelector(state),
+});
+
+const MAX_ITEMS_PER_PAGE = 50;
+const AnalysisPillarRenderer = (props: AnalysisPillarRendererProps) => {
+    const {
+        pillarId,
+        title,
+        assigneeName,
+        setDeleteId,
+    } = props;
+
+    const handleDeletePillar = useCallback(() => {
+        setDeleteId(pillarId);
+    }, [pillarId]);
```

Comment:
here setDeleteId is a function passed to the component. I needs to be added to the dependency array for handleDeletePillar hook

Reply:
I second!